whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapRelationService.java

```diff
@@ -270,4 +272,12 @@ private String objectTypesToString(final Collection<ObjectType> objectTypes) {
         return COMMA_JOINER.join(objectTypes.stream().map(ObjectType::getName).toList());
     }
 
+    private IpInterval transformToIpInterval(final Interval interval) {
```

Comment:
This method is unnecessary, please remove it

Reply:
The method is not doing anything. You are passing an `IpInterval` and returning an `IpInterval`

Reply:
no, I am passing `Interval` and returning `IpInterval`

Reply:
All you are doing is casting the parameter and returning it, you're not transforming it. I suggest merging it into the other method.

Reply:
OK, I merged that into the calling methods

Reply:
OK I see what you're doing, the method *is* necessary but please rename it from `transformToIpRangeString` to `intervalToString` for consistency (i.e. see the preceding method). Also restore the `transformToIpInterval` method as it's more readable but rename to `internalToIpInterval`.

Reply:
It's possible to "fix" this so it's unnecessary, by making `Interval` and `IpInterval` `sealed` and only allowing `Ipv4Resource` and `Ipv6Resource` as implemented classes.

Reply:
Done by your PR and updated this one which those changes